Dapper/SqlMapper.cs

```diff
@@ -2533,9 +2533,16 @@ internal static IList<LiteralToken> GetLiteralTokens(string sql)
             }
 
             bool filterParams = false;
-            if (removeUnused && identity.commandType.GetValueOrDefault(CommandType.Text) == CommandType.Text)
+            if (Settings.OleDbCheckEnabled)

```

Comment:
it is important that we don't change the behaviour in unrelated cases; this *looks* like it is now going to filter sprocs, which would be very bad; suggest simplifying to just:

``` diff
- filterParams = !smellsLikeOleDb.IsMatch(identity.sql);
+ filterParams = !(Settings.OleDbCheckEnabled && smellsLikeOleDb.IsMatch(identity.sql);
```

(please double-check my boolean logic)

Reply:
Even in the case of legacy parameter tokens, shouldn't the regex be evaluated only in the case of `CommandType.Text`

Reply:
yes, but I was referring to the value of `filterParams` - with this setting disabled, it would have changed from `false` to `true` in the original version, no?

Reply:
It would filter for sprocs too which is now fixed.